packages/loot-design/src/components/modals/ImportTransactions.js

```diff
@@ -40,37 +40,48 @@ function parseDate(str, order) {
     return v && v.length === 1 ? '0' + v : v;
   }
 
-  let parts = str.replace(/ /g, '').split(/[^0-9]/);
+  const dateGroups = (a, b) => str => {
+    const digits = str.replace(/[^\d]/g, '');
+    return [digits.slice(0, a), digits.slice(a, a + b), digits.slice(a + b)];
+  };
+  const yearFirst = dateGroups(4, 2);
+  const twoDig = dateGroups(2, 2);
 
-  let year, month, day;
+  let parts, year, month, day;
   switch (order) {
```

Comment:
Alternatively, we could apply the groups of two version as a default and use the `(4,2)` version for `yyyy mm dd`, as it's the only format that actually needs something different.
```js
  let parts = dateGroups(2, 2)(str);

  let year, month, day;
  switch (order) {
    case 'dd mm yyyy':
      year = parts[2];
```
```js
// ...
    case 'yyyy mm dd':
      parts = dateGroups(4, 2)(str);
      year = parts[0];
      month = parts[1];
      day = parts[2];
      break;
```
The `dateGroups` function could be dropped, or not curried, etc.

Reply:
I think this is good enough for now! I'll go ahead and merge this to speed things along, but if you want to open any follow-up PRs to keep working on that code that works! thanks!